plugins/database/src/general_query.cpp

```diff
@@ -2206,6 +2206,7 @@ checkCondInputAccess( genQueryInp_t genQueryInp, int statementNum,
         else {
             zoneName = genQueryInp.condInput.value[zoneIx];
         }
+rodsLog(LOG_NOTICE, "XXXX %s:%d", __FILE__, __LINE__);
```

Comment:
remove debug

Reply:
Previous there was no recursive behavior.  The original query worked as intended, the only ticket applied to a collection of data objects was a ticket of the immediate parent.  This commit provides the desired recursive behavior by modifying the query and walking the collection path.